packages/flutter/lib/src/foundation/platform.dart

```diff
@@ -55,12 +72,18 @@ TargetPlatform get defaultTargetPlatform {
   }
   return result;
 }
+
 /// Override the [defaultTargetPlatform].
 ///
 /// Setting this to null returns the [defaultTargetPlatform] to its original
 /// value (based on the actual current platform).
 ///
-/// This setter is only available intended for debugging purposes. To change the
+/// Generally speaking this override is only useful for tests. To change the
 /// target platform in release builds, use the platform override APIs (such as
 /// [ThemeData.platform] in the material library).
```

Comment:
Perhaps this sentence should be tweaked as well? Setting ThemeData.platform doesn't **actually** change the target platform, per discussion, just whether visual theming uses it.